apps/backend/test/unit/deposits/cash-deposit.service.spec.ts

```diff
@@ -86,119 +76,229 @@ describe('CashDepositService', () => {
-        { ...cashDepositMock, status: MatchStatus.MATCH }
+      const updatedEntities = await cashDepositService.updateDeposits(
+        cashDeposits
+      );
+
+      expect([updatedEntities[0], updatedEntities[1]]).toEqual([
+        cashDeposits[0],
+        cashDeposits[1]
       ]);
     });
   });
 
-  describe('findDistinctDepositDates', () => {
-    it('should return all distinct deposit dates', async () => {
-      const spy = jest
-        .spyOn(service, 'findCashDepositDatesByLocation')
-        .mockResolvedValue(['2020-01-01', '2020-01-02']);
-      const params = {
-        program: Ministries.SBC,
-        dateRange: {
-          to_date: '2020-01-02',
-          from_date: '2020-01-01'
-        },
-        location: locationMock
-      };
-      expect(
-        service.findCashDepositDatesByLocation(
-          params.program,
-          params.dateRange,
-          params.location
-        )
-      ).resolves.toEqual(['2020-01-01', '2020-01-02']);
-      expect(spy).toBeCalledTimes(1);
+  describe('updateDeposit', () => {
+    it('should update and return a single CashDepositEntity instance', async () => {
+      const expected = cashDeposits[0];
+      expected.status = MatchStatus.MATCH;
+
+      jest
+        .spyOn(cashDepositRepo, 'save')
+        .mockResolvedValueOnce(expected as any);
+
+      const updatedEntity = await cashDepositService.updateDeposit(
+        cashDeposits[0]
+      );
+
+      expect(updatedEntity).toEqual(expected);
     });
   });
 
-  describe('findCashDepositsByDateLocationAndProgram', () => {
-    it('should return all cash deposits for a given date, location and program', async () => {
-      const spy = jest
-        .spyOn(service, 'findCashDepositsByDate')
-        .mockResolvedValue([cashDepositMock]);
-      const params = {
-        program: Ministries.SBC,
-        dateRange: {
-          to_date: '2020-01-02',
-          from_date: '2020-01-01'
-        },
-        location: locationMock
-      };
-      expect(
-        service.findCashDepositsByDate(
-          params.program,
-          params.dateRange.to_date,
-          params.location
-        )
-      ).resolves.toEqual([cashDepositMock]);
-      expect(spy).toBeCalledTimes(1);
+  describe('findCashDepositExceptions', () => {
+    it('should return an array of CashDepositEntity instances matching the given parameters', async () => {
+      const program = Ministries.SBC;
+      const depositDate = '2022-01-01';
+      const location = { pt_location_id: 1 } as LocationEntity;
+
+      const expectedEntities = [
+        { id: 1, deposit_amt_cdn: 100 },
+        { id: 2, deposit_amt_cdn: 200 }
+      ];
+
+      jest
+        .spyOn(cashDepositRepo, 'find')
+        .mockResolvedValueOnce(expectedEntities as any);
+
+      const foundEntities = await cashDepositService.findCashDepositExceptions(
+        depositDate,
+        program,
+        location
+      );
+
+      expect(foundEntities).toEqual(expectedEntities);
```

Comment:
I like the idea behind this test, but I'm not sure there's a failure case. `find` literally just returns specific values, so the cases in `findCashDepositExceptions` (like the status needing to be In Progress) isn't actually tested. I think ultimately, it's the difficult part of relying on mocks for tests.

Need to think about / discuss this.

Reply:
yes, I agree this test is not very useful. I think the cash-deposit service and pos-deposit service would be better tested as part of an e2e testing, since they're primarily just a pass through service to connect to the database. 

I'm definitely open to trying to create some more robust unit tests here, but there isn't a lot of logic to test as most of it is handled by typeorm... I'm definitely open to any ideas/suggestions on creating better tests for these services though!